### PR TITLE
tests, net, sriov: remove use GetRunningVMIDomainSpec

### PR DESCRIPTION
**Pull request by nirdothan**
End to end tests are intended for high level functionality validation.
`GetRunningVMIDomainSpec` is a function that parses and checks the `DomainXML` details, which is an intermediate tech detail that could be tested in Unit Tests but does not belong in e2e tests.
In addition, GetRunningVMIDomainSpec is implemented in from test.utils, which is a central module external to network-sig.  We strive to decouple network logic and tests from other areas in the code and generally to eliminate test.utils alltogther.
Given the above, this PR removes one use of `GetRunningVMIDomainSpec`, additional removal PRs will follow.

### What this PR does
Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/network/sriov.go [hunk 1]
```diff
@@ -172,36 +172,28 @@ var _ = Describe("SRIOV", Serial, decorators.SRIOV, func() {
 			if vmi.Annotations == nil {
 				vmi.Annotations = make(map[string]string)
 			}
+			const (
+				specialNetTag = "specialNet"
+				pciAddress    = "0000:08:00.0"
+			)
 			vmi.Annotations[v1.InstancetypeAnnotation] = testInstancetype
```
Comment (orelmisan) [inline tests/network/sriov.go:179]: Please consider moving the consts definition bellow the annotation assignment, as it breaks an existing logical flow:
```suggestion
			vmi.Annotations[v1.InstancetypeAnnotation] = testInstancetype
			const (
				specialNetTag = "specialNet"
				pciAddress    = "0000:08:00.0"
			)
```
BTW, the annotation definition could be replaced by `libvmi.WithLabel`, thus removing additional logic from the actual test.

File: tests/network/sriov.go [hunk 2]
```diff
@@ -172,36 +172,28 @@ var _ = Describe("SRIOV", Serial, decorators.SRIOV, func() {
...
 
 			for idx, iface := range vmi.Spec.Domain.Devices.Interfaces {
 				if iface.Name == sriovnet1 {
-					iface.Tag = "specialNet"
+					iface.Tag = specialNetTag
+					iface.PciAddress = pciAddress
 					vmi.Spec.Domain.Devices.Interfaces[idx] = iface
 				}
 			}
```
Comment (orelmisan) [inline tests/network/sriov.go:187]: Please consider replacing this loop with an interface definition on the vmi specification above.

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @nirdothan.
I find "testing the system end to end (and not looking at the domain which is an implementation detail)", another (stronger IMO) reasoning for performing this change.

**Review by orelmisan — APPROVED** [review]
Thank you

**Review by EdDev — COMMENTED** [review]
Looks great, thanks!

Please fix the commit message for fit the recommended line length [1].

[1] https://cbea.ms/git-commit/#wrap-72

**Review by orelmisan — APPROVED** [review]

**Pull request merged**